### PR TITLE
git: Ensure the tagging script is idempotent

### PR DESCRIPTION
**Pull request by jumanjiman**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -12,6 +12,9 @@ end_of_line = lf
 insert_final_newline = true
 trim_trailing_whitespace = true
 
+[*.sh]
+indent_style = tab
```
Comment (purpleidea) [inline .editorconfig]: oh wow. <3

File: tag.sh [hunk 2]
```diff
@@ -1,14 +1,50 @@
 #!/bin/bash
-# TODO: don't run if current HEAD is already tagged (ensure this is idempotent)
-# take current HEAD with new version
+set -eEu
```
Comment (purpleidea) [inline tag.sh]: I didn't even know there were this many options! ;)

**Comment by purpleidea** [comment]
Merged, thank you again!

**Comment by purpleidea** [comment]
@jumanjiman Since you seem to like shell, perhaps you'd like this: https://github.com/purpleidea/mgmt/issues/286 ?

**Comment by jumanjiman** [comment]
> Since you seem to like shell, perhaps you'd like this: #286

yep, i'll take it.

File: tag.sh (hunk 2 above)
Comment (jumanjiman) [reply tag.sh]: things one learns with shellcheck and style guides :-)

**Pull request merged**